post/webnote-extension/index.md

```diff
@@ -0,0 +1,658 @@
+        <button style="display: block; margin-top: 16px" @click="save">
+            👍 Done
+        </button>
+    </div>
+</template>
+
+<script>
+    import { createItem, readItem, updateItem } from "@directus/sdk";
+    export default {
+        inject: ["directus"],
+        data() {
+            return {
+                note: "",
+            };
+        },
+        computed: {
+            id() {
+                return this.$route.params.id;
+            },
+            isCreate() {
+                return this.$route.params.id === "+";
+            },
+            isEdit() {
+                return !this.isCreate;
+            },
+        },
+        created() {
+            if (this.isEdit) {
+                this.get();
+            }
+        },
+        methods: {
+            async get() {
+                // Edit mode
+                const { note } = await this.directus.request(
+                    readItem("notes", this.id)
+                );
+                this.note = note;
+            },
+            async save() {
+                if (this.isEdit) {
+                    await this.directus.request(
+                        updateItem("notes", this.id, {
+                            note: this.note,
+                        })
+                    );
+                } else {
+                    const [tab] = await chrome.tabs.query({
+                        active: true,
+                        lastFocusedWindow: true,
+                    });
+
+                    const { origin } = new URL(tab.url);
+                    await this.directus.request(
+                        createItem("notes", {
+                            note: this.note,
+                            website: origin,
+                        })
+                    );
+                }
+
+                this.$router.push({ name: "home" });
+            },
+        },
+    };
+</script>
+```
+
+### Load `upsert.vue` in `router.js`
+
+```js
+import Upsert from "../views/upsert.vue";
+```
+
+```js
+    {
+        path: "/note/:id",
+        name: "upsert",
+        meta: { public: false },
+        component: Upsert,
+    },
+```
+
+![Create Note](./images/create.png)
+![Edit Note](./images/edit.png)
+
+## Summary
+
+Directus WebNote Keeper is a proposed Chrome extension designed to help users save and manage notes related to the webpages they visit. It leverages Directus as the backend CMS for storing and managing these notes. When users click the extension while browsing, it automatically captures the current webpage's URL and prompts the user to add a note. Users can then view, edit, and delete their notes directly from the extension's interface. The extension will use the Directus API to perform CRUD (Create, Read, Update, Delete) operations on the notes. This project integrates the functionalities of a headless CMS with Chrome extension development to provide a practical tool for organizing web research and personal thoughts.
```

Comment:
```suggestion
In this tutorial, you've learnt how to build a Chrome Extension that authenticates with Directus and allows the user to manage data. There's still some more polish and functionality you can build, but a lot of it will be based on the same concepts we've worked through here.
```